### ISSUE TITLE
Feature request: `no-nth-methods` option to allow positional selection bound to an ordinal name

### ISSUE THREAD

**Issue by neupenguin** [playwright-community/eslint-plugin-playwright#511]
## The problem

`no-nth-methods` reports every `first()`, `last()` and `nth()`. That is the right default: selecting by position is fragile when the test does not care about position, and Playwright's own docs steer away from it.

But some tests are about position. They verify an order or a subset:

```ts
// Newest investigation is listed first
await expect(rows.first()).toHaveText('checkout latency')
await expect(rows.nth(1)).toHaveText('db replication lag')
```

Here there is nothing else to locate by. The position is the assertion. With the rule on, a suite has three choices, all worse:

1. **Disable the rule per line.** This works, but every order test ends up with a disable comment, and nothing separates a deliberate positional check from a lazy one.
2. **Work around it.** Read `allTextContents()` into an array and index that. The positional check survives, but it loses auto-waiting and web-first assertions, which is worse than the code the rule rejected.
3. **Turn the rule off.** Then the lazy `page.locator('button').first().click()` is back.

What separates a deliberate positional check from a lazy one is whether the code says which position it means. `rows.first().click()` doesn't. `const firstRow = rows.first()` does, and the assertion then reads as the order it verifies.

## Proposal

Add an option to `no-nth-methods` that allows a positional call when it initializes a variable named for the position:

```json
{
  "playwright/no-nth-methods": ["error", { "allowOrdinalNames": true }]
}
```

With the option on, a call is allowed only when all of these hold:

- it is the direct initializer of a variable declarator, `const <name> = <locator>.<method>(…)`;
- `<name>` starts with the ordinal the call selects, followed by an uppercase letter:
  - `first()` → `first…`
  - `last()` → `last…`
  - `nth(k)` with a numeric literal → the ordinal of `k + 1`, so `nth(0)` → `first…`, `nth(1)` → `second…`, `nth(2)` → `third…`, and so on
  - `nth(expr)` with a computed index → `nth…` or `last…`, because a word cannot be checked against an expression

Everything else is still reported.

Examples of **correct** code with `{ "allowOrdinalNames": true }`:

```ts
const firstButton = page.getByRole('button').first()
const secondValue = values.nth(1)
const lastRow = rows.last()
const nthRow = rows.nth(i)
const lastOption = options.nth(count - 1)

await expect(firstButton).toHaveText('Save')
await expect(secondValue).toHaveText('42')
```

Examples of **incorrect** code with `{ "allowOrdinalNames": true }`:

```ts
await rows.first().click()                   // not bound to a name
expect(rows.nth(2)).toHaveText('x')          // not bound to a name
const row = rows.first()                     // name does not state the position
const first = rows.first()                   // ordinal with no noun
const firstValue = values.nth(1)             // nth(1) is the second, not the first
const lastRow = rows.first()                 // name contradicts the call
const secondRow = rows.nth(i)                // computed index cannot be checked as "second"
let firstRow; firstRow = rows.first()        // assignment, not initialization
```

The rule can't know whether a test really verifies order. The name is the part it can check, and it is what a reviewer reads, so a positional selection has to be stated rather than happen by accident.

### Default

Off (`false`), so current behaviour doesn't change.

### Message

A report under the option can say what is expected, e.g.:

> `first()` selects by position. Use it only to verify an order or a subset, and bind it to a variable named for the position, such as `firstRow`; otherwise locate by role, name or test id.

## Reference implementation

We run this today as a local rule next to the plugin; happy to open a PR folding it into `no-nth-methods`.

```js
const ORDINALS = ['first', 'second', 'third', 'fourth', 'fifth', 'sixth', 'seventh', 'eighth', 'ninth', 'tenth']

export default {
  meta: {
    type: 'problem',
    schema: [],
    messages: {
      unnamed:
        '`{{method}}()` selects by position. Use it only to verify an order or a subset, and bind it to a variable named for the position, such as `{{example}}`; otherwise locate by role, name or test id.',
    },
  },
  create(context) {
    return {
      CallExpression(node) {
        const { callee } = node
        if (callee.type !== 'MemberExpression' || callee.computed) return
        const method = callee.property.name
        if (method !== 'first' && method !== 'last' && method !== 'nth') return

        const [index] = node.arguments
        let prefixes
        if (method !== 'nth') prefixes = [method]
        else if (index?.type === 'Literal' && ORDINALS[index.value]) prefixes = [ORDINALS[index.value]]
        else prefixes = ['nth', 'last']

        const { parent } = node
        const name =
          parent.type === 'VariableDeclarator' && parent.init === node && parent.id.type === 'Identifier'
            ? parent.id.name
            : ''
        if (prefixes.some((prefix) => new RegExp(`^${prefix}[A-Z]`).test(name))) return

        context.report({
          node: callee.property,
          messageId: 'unnamed',
          data: { method, example: `${prefixes[0]}Row` },
        })
      },
    }
  },
}
```

Folded into the plugin, it would use the existing locator detection in `no-nth-methods` rather than matching method names alone, and it would only take the naming path when the option is on.

### Open questions

- **Ordinals past `tenth`:** they could fall back to `nth…`, or stay unsupported. An order test rarely checks beyond the first few positions.
- **Destructuring:** `const [firstRow] = …` doesn't apply, since locators are not iterables. Left out.
- **Chained locators:** `const firstCell = rows.first().getByRole('cell')` is reported today because the positional call is not the initializer. Allowing a chain whose root is the positional call would be a reasonable extension.
